### PR TITLE
Fully drop dependency on libfm C library.

### PR DESCRIPTION
**Pull request by PCMan**
Finally, the last piece to the full C++ port.
C code that is hard to port is copied and put into src/core/legacy.
Gio VFS implementations for menu:// and search:// are sitll in C and put in src/core/vfs.
Cheers!

This definitely requires much testing, though.

### CODE REVIEW

**Comment by PCMan** [comment]
@tsimonq2 With this change, libfm is no longer a dependency of libfm-qt. So you will need to update your deb package for the next libfm-qt release. :-)

**Comment by tsujan** [comment]
@PCMan Thanks you very much!

**Comment by PCMan** [comment]
@tsujan yes I know. Let's merge #195 first, and I'll rebase this one on the new master.

**Comment by agaida** [comment]
BTW: this so-name thing is the reason why the new libfm-qt and dependend packages are not in sid yet - they are in experimental, but the builders for experimental run with far lower priority - and that means - hopefully the packages for the release arches are built within the next two days, then i can add/fix the changed symbols and migrate libfm-qt/lxqt-qtplugin/pcmanfm-qt/lximage-qt to sid

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -126,7 +137,10 @@ install(EXPORT
 target_link_libraries(${LIBFM_QT_LIBRARY_NAME}
     Qt5::Widgets
     Qt5::X11Extras
-    ${FM_LIBRARIES}
+    ${GLIB_LIBRARIES}
```
Comment (palinek) [inline src/CMakeLists.txt]: Why aren't we using the `*_LDFLAGS` ? This caused problems with libraries installed in non-standard paths problems in other components in the past...

Comment (PCMan) [reply src/CMakeLists.txt]: @palinek our FindGLIB.cmake does not define LDFLAGS:
https://github.com/lxqt/lxqt-build-tools/blob/master/cmake/find-modules/FindGLIB.cmake

Comment (palinek) [reply src/CMakeLists.txt]: Sorry,... forgot about that... I thought we're using just the pkg-config...

**Comment by luis-pereira** [comment]
Building and testing it.

**Comment by tsujan** [comment]
I started to use this and https://github.com/lxqt/pcmanfm-qt/pull/684. So far so good. Will report probable issues here.

Of course, it needs rebasing.

**Comment by PCMan** [comment]
rebased on the current master.

**Pull request merged**